### PR TITLE
[path_provider_macos] Adds example app

### PR DESCRIPTION
**Pull request by franciscojma86**
## Description

This is a copy of the core plugin's test app, but removed the platform conditions, and just kept the directories supported on macOS.

## Related Issues

https://github.com/flutter/flutter/issues/46307

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/path_provider/path_provider_macos/example/test_driver/path_provider_e2e.dart [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  });
+
+  testWidgets('getExternalStorageDirectory', (WidgetTester tester) async {
+    if (Platform.isIOS) {
+      final Future<Directory> result = getExternalStorageDirectory();
+      expect(result, throwsA(isInstanceOf<UnsupportedError>()));
+    } else if (Platform.isAndroid) {
+      final Directory result = await getExternalStorageDirectory();
+      _verifySampleFile(result, 'externalStorage');
+    }
```
Comment (ditman) [inline packages/path_provider/path_provider_macos/example/test_driver/path_provider_e2e.dart]: Is this run against all platforms? Shouldn't this test what happens on the macos platform?

Comment (franciscojma86) [reply packages/path_provider/path_provider_macos/example/test_driver/path_provider_e2e.dart]: nope, good catch :)

File: packages/path_provider/path_provider_macos/example/test_driver/path_provider_e2e.dart [hunk 2]
```diff
@@ -36,55 +36,6 @@ void main() {
       expect(result, throwsA(isInstanceOf<UnsupportedError>()));
```
Comment (ditman) [inline packages/path_provider/path_provider_macos/example/test_driver/path_provider_e2e.dart]: (This is also a platform-dependent test)

**Comment by ditman** [comment]
I've seen this error happening in master as well:

```
RUNNING FIREBASE TEST LAB TESTS for path_provider/path_provider
Activated service account credentials for: [cirrus-firebase-testlab@flutter-infra.iam.gserviceaccount.com]
Updated property [core/project].
ERROR: (gcloud.config.set) timed out
This may be due to network connectivity issues. Please check your network settings, and the status of the service you are trying to reach.
����
```

**Comment by franciscojma86** [comment]
They passed :) thanks!

**Pull request merged**